qiskit_nature/problems/sampling/protein_folding/interactions/miyazawa_jernigan_interaction.py

```diff
@@ -0,0 +1,47 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""A class defining a Miyazawa-Jernigan interaction between beads of a peptide."""
+from typing import List
+
+import numpy as np
+from .interaction import Interaction
+from ..residue_validator import _validate_residue_sequence
+from ..data_loaders.energy_matrix_loader import (
+    _load_energy_matrix_file,
+)
+
+
+class MiyazawaJerniganInteraction(Interaction):
+    """A class defining a Miyazawa-Jernigan interaction between beads of a peptide."""
+
+    def calculate_energy_matrix(self, chain_len: int, residue_sequence: List[str]) -> np.ndarray:
```

Comment:
I think I commented on this a while back but I don't exactly recall: what is the benefit/need for passing in `chain_len` as well as `residue_sequence`? Can't you infer the former from the latter? I simply want to clarify this again.

Reply:
You are right that one can infer the other in the way is done, but for consistency with the problem definition and as an extra check we can keep it as is. I leave the decision to @dlasecki !

Reply:
To make it clearly visible what the size of a chain is instead of counting the number of terms in the list of residues and to cross-validate later with lengths of side chain residue list and main chain residue list. I could imagine a user defines both residue lists that have a length one shorter than desired and they don't notice they deal with a different peptide because of that.
If you think I am too pessimistic then I could get rid of this and infer.

Reply:
Okay no it's fine to keep it explicit. However, Panos mentioned an extra check to ensure consistency: I did not notice this anywhere. Did I overlook that?

Reply:
this is the way we define the protein folding problem, possibly I overused the word consistency, but this the way we input things in the protein folding folding!

Reply:
I mean that does not really count as an argument though, does it? You could also remove the `chain_len` in the Problem for the exact same reason (that is can be inferred).
But I agree with Dariusz that it helps with readability to specify the length separately from the list. However, this is only true if the list and the `chain_len` actually get checked for consistency. Otherwise the user won't get informed about a potential mismatch.

So my question is: does such a check occur?

Reply:
Right, I did not check all the combinations. Thanks.